quicklogic/common/utils/process_sdc_constraints.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+"""
+This script processes an SDC constraint file and replaces all references to
+pin names in place of net names with actual net names that are mapped to those
+pins. Pin-to-net mapping is read from a PCF constraint file.
+
+Note that this script does not check whether the net names in the input PCF
```

Comment:
If user provides a pin name that is not mentioned in PCF then the script will report an error.

If user provides an invalid pin name (i.e. one that is not present in the pinmap) it just won't be translated to a net and will be left as-is. Later VPR will fail complaining that there is no net of that name. Since the pin name is invalid the placement constraints utility script will fail too.

In a more complicated scenario: A pin name in SDC is correct but it is mapped to a non-existing net in PCF. The SDC processing script will map it according to the PCF so in the end the name of the non-existing net will end up in SDC. In this case VPR will fail too for the same reason.

A more comprehensive check can be added to the SDC processing script - it would require it to read eg. EBLIF netlist and identify all existing net names so that it knows what the correct names are. Shall we do that?

Reply:
I think it is better if we can do the checks by looking at design file as well as pin-mapping csv file to validate the  SDC net/pin inputs.

Reply:
Ok, I'll look into implementing that.